### PR TITLE
Feat/product endpoints

### PR DESCRIPTION
**Pull request by tobisupreme**
This pull request is a work in progress

The controllers for `createProduct` and `updateProduct` are "finished".
My colleague has "completed" the code for `getProductById` and `deleteProduct`. (They will be opening pull requests to that regard)

I am currently in the process of writing the code for `getAllProducts`.

Pending when subsequent commits are ready, we are open to reviews, feedback, comments, etc 🙏

### CODE REVIEW

**Comment by tobisupreme** [comment]
#4

File: products/productController.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+const productModel = require('../model/products')
+
+const createProduct = async (req, res, next) => {
+  try {
+    /**
+     * create new product with required parameters
+     */
+    const productToSave = new productModel({
```
Comment (Skywonda) [inline products/productController.js:8]: make basic validations on the data types you are collecting before saving them to the database

File: products/productController.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  } catch (error) {
+    /**
+     * In case of error
+     * send to error handling middleware
+     */
+    next(error)
+  }
+}
+
+const updateProduct = async (req, res, next) => {
```
Comment (Skywonda) [inline products/productController.js]: add this routers into the route files

**Review by Skywonda — DISMISSED** [review]
do some validations and then write the put the router in the router file

**Comment by tobisupreme** [comment]
Hi boss! I resolved some conflicts and updated this pull request 🙏

**Pull request merged**